packages/lib/src/Deferred.ts

```diff
@@ -0,0 +1,43 @@
+export default class Deferred<T> {
+  private _promise: Promise<T>;
+
+  private _resolve: (val: T | PromiseLike<T>) => void;
+
+  private _reject: (reason?: any) => void;
+
+  constructor() {
+    this._resolve = () => {};
+    this._reject = () => {};
+
+    this._promise = new Promise((resolve, reject) => {
+      this._resolve = resolve;
+      this._reject = reject;
+    });
+  }
+
+  resolve(value: T) {
+    this._resolve(value);
+    return this;
+  }
+
+  reject(value: any) {
+    this._reject(value);
+    return this;
+  }
+
+  promise() {
+    return this._promise;
+  }
+
+  then: Promise<T>['then'] = (...args) => {
+    return this._promise.then(...args);
+  };
+
+  catch: Promise<T>['catch'] = (...args) => {
```

Comment:
Hmm... This approach seems a bit heavy-handed, but I can't think of a much better way to go about it.

That said, since we aren't using `then` or `catch` anywhere, is there a reason you wanna keep them around?

Additionally, we might consider moving this out of `lib` somewhere. Right now, there isn't a good way to tree-shake dependencies for our packages, and consumers wouldn't be using this, we would currently be including this in every app using `leafygreen-ui` since pretty much all of our components include this package. I'm working on the tree-shaking issue sooner-than-later, but for the time being, it's still an issue.